subprojects/core/src/main/java/org/gradle/api/internal/file/archive/TarFileTree.java

```diff
@@ -110,21 +113,71 @@ private File getBackingFile() {
         if (tarFile != null) {
             return tarFile;
         }
-        if (resource != null) {
-            return resource.getBackingFile();
+        return resource.getBackingFile();
+    }
+
+    private File getExpandedDir() {
+        InputStream inputStream = null;
+        try {
+            inputStream = new BufferedInputStream(resource.read());
```

Comment:
I basically copied the logic from `visit()`. Good to know it's not needed here. It should be documented though. I'll add a short sentence to the javadoc.